fix this @copilot 
1s
Run pytest
  pytest
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.12.13/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.13/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.13/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.13/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.13/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.13/x64/lib
============================= test session starts ==============================
platform linux -- Python 3.12.13, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/runner/work/smart-google/smart-google
collected 0 items / 1 error

==================================== ERRORS ====================================
______________________ ERROR collecting tests/test_app.py ______________________
ImportError while importing test module '/home/runner/work/smart-google/smart-google/tests/test_app.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/test_app.py:13: in <module>
    from app import app as flask_app  # noqa: E402
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app.py:7: in <module>
    from flask import Flask, send_from_directory
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/flask/__init__.py:14: in <module>
    from jinja2 import escape
E   ImportError: cannot import name 'escape' from 'jinja2' (/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jinja2/__init__.py)
=========================== short test summary info ============================
ERROR tests/test_app.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.16s ===============================
Error: Process completed with exit code 2.

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,10 +1,10 @@
 gunicorn==25.1.0
-Flask==1.1.2
+Flask>=2.0.3,<3.0.0
 Flask-MQTT==1.1.1
 Flask-OAuthlib==0.9.6
-Flask-SQLAlchemy==2.5.1
-SQLAlchemy==1.4.47
-Werkzeug>=2.0.0,<2.4.0
+Flask-SQLAlchemy>=3.0.0,<4.0.0
+SQLAlchemy>=1.4.18,<3.0.0
+Werkzeug>=2.3.7,<3.0.0
 Jinja2>=3.0.0,<4.0.0
 MarkupSafe>=2.0.0,<3.0.0
 itsdangerous>=2.0.0,<3.0.0
@@ -16,4 +16,4 @@ requests-oauthlib
 requests>=2.31.0
 cryptography>=41.0.0
 python-dotenv==1.0.0
-Flask-Login==0.4.1
+Flask-Login>=0.6.0,<1.0.0

